Setup script not resolving components/api-server/package.json

Running ```make setup``` throws errors when trying to ```npm install```:

* At first, npm install failed because dev-site/package.json was referencing the production branch of service-core, which is not existing anymore :
>     "pryv-service-core": "git+ssh://git@github.com:pryv/service-core.git#production"
We patched this in dev-site/master by referencing service-core/release-1.1 instead.

* Now, npm install still fails with the following error:
```
Installing Node modules from 'package.json' if necessary...

npm ERR! code ENOLOCAL
npm ERR! Could not install from "node_modules/pryv-service-core/components/api-server" as it does not contain a package.json file.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/thiebaudmodoux/.npm/_logs/2017-11-27T15_58_02_648Z-debug.log
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,7 @@
 .DS_Store
 .project
 build
+dependencies
 dev-site.iml
 node_modules
 npm-debug.log

--- a/README.md
+++ b/README.md
@@ -7,7 +7,7 @@ Source content for api.pryv.com (API reference, recommendations, guides, etc.)
 
 ### Prerequisites
 
-Node v4.4.3+.
+Node v8.3.0+, Yarn v0.21.3+
 
 `make setup` sets up the environment.
 

--- a/README.md
+++ b/README.md
@@ -11,9 +11,11 @@ Node v4.4.3+.
 
 `make setup` sets up the environment.
 
-### Build & publish
+**Note:** the setup command will install a dependency to pryv/service-core.
+By default, it will target the branch defined inside the setup script.
+If you want to use another branch, use `make setup core=my_branch` instead.
 
-Prerequisite: the source API server working copy must be under `../api-server`.
+### Build & publish
 
 - `make build` generates the website from the source into the `build` folder
 - `make watch` watches the source folder and rebuilds on changes

--- a/dev-env/setup.sh
+++ b/dev-env/setup.sh
@@ -10,16 +10,37 @@ cd $scriptsFolder/..
 hash git 2>&- || { echo >&2 "I require git."; exit 1; }
 hash yarn 2>&- || { echo >&2 "I require node and yarn."; exit 1; }
 
-echo "
-Installing Node modules from 'package.json' if necessary...
-"
+# resolve service-core dependency
+if [ ! -d dependencies/core ]
+then
+  echo "Setting up 'dependencies/core' folder for service-core dependency."
+  git clone git@github.com:pryv/service-core.git dependencies/core
+fi
+
+if [ -z "$1" ]
+then
+  # default branch used for service-core dependency
+  coreBranch="release-1.1"
+else
+  coreBranch=$1
+fi
+
+echo "Service-core dependency is targeting the following branch: $coreBranch"
+
+# ensure service-core dependency is up-to-date
+cd dependencies/core
+git checkout $coreBranch
+git pull
+cd $scriptsFolder/..
+
+# install node modules
+echo "Installing Node modules from 'package.json' if necessary."
 yarn install
 
+# setup build folder
 if [ ! -d build ]
 then
-  echo "
-Setting up 'build' folder for publishing to GitHub pages...
-"
+  echo "Setting up 'build' folder for publishing to GitHub pages."
   git clone git@github.com:pryv/pryv.github.io.git build
 fi
 

--- a/dev-env/setup.sh
+++ b/dev-env/setup.sh
@@ -20,7 +20,7 @@ fi
 if [ -z "$1" ]
 then
   # default branch used for service-core dependency
-  coreBranch="release-1.1"
+  coreBranch="release-1.2"
 else
   coreBranch=$1
 fi

--- a/makefile
+++ b/makefile
@@ -13,7 +13,7 @@ clean:
 	rm -rf build/*
 
 setup:
-	./dev-env/setup.sh
+	./dev-env/setup.sh $(core)
 
 retrieve-types:
 	@echo ""

--- a/package.json
+++ b/package.json
@@ -34,7 +34,7 @@
     "metalsmith-stylus": "^2.0.0",
     "metalsmith-templates": "^0.7.0",
     "metalsmith-watch": "^1.0.1",
-    "pryv-service-core": "git+ssh://git@github.com:pryv/service-core.git#release-1.1",
+    "pryv-service-core": "file:dependencies/core",
     "pryv-style": "git://github.com/pryv/style.git",
     "unix-timestamp": "^0.2.0",
     "rec-la": "git+https://github.com/pryv/rec-la.git"

